dask/dataframe/shuffle.py

```diff
@@ -348,9 +348,12 @@ def rearrange_by_column(
     if shuffle == "disk":
         return rearrange_by_column_disk(df, col, npartitions, compute=compute)
     elif shuffle == "tasks":
-        return rearrange_by_column_tasks(
+        df2 = rearrange_by_column_tasks(
             df, col, max_branch, npartitions, ignore_index=ignore_index
         )
+        if ignore_index:
+            df2._meta = df2._meta.reset_index(drop=True)
+        return df2
```

Comment:
Any reason to do this here rather than in `rearrange_column_by_tasks`?

Reply:
Good question -It seemed to me that I would need to add this in two places in `rearrange_by_column_task` (since the `_simple_rearrange_by_column_tasks` code path could be used), but only once in `rearrange_by_column`.

Reply:
Is this reasoning sufficient, or should I move the changes down into rearrange_by_column_task?

Reply:
@TomAugspurger -- just a gentle ping on this